containerbuddy/config.go

```diff
@@ -311,8 +313,19 @@ func initializeConfig(config *Config) (*Config, error) {
 			return nil, ifaceErr
 		}
 
-		if service.ipAddress, err = GetIP(interfaces); err != nil {
-			return nil, err
+		if service.Endpoint == "" && service.ipAddress ==  "" {
+			if service.ipAddress, err = GetIP(interfaces); err != nil {
+				return nil, err
+			}
+		} else {
+			if ok := net.ParseIP(service.Endpoint); ok == nil {
+				if ip, err := net.LookupHost(service.Endpoint); err == nil {
```

Comment:
Wanted to add here that the `ipAddress` field is just a string, and that's what we're giving to Consul in the `/v1/agent/service/register/` call. If we're allowing hostnames at all, why not just pass them through as-is to Consul rather than trying to look them up here. Doing a lookup is a misleading validation, because it can go away afterwards. And Containerbuddy doesn't do anything with the addresses other than give them to Consul anyways, whereas if you're going down this path then presumably your application has some mechanism to deal with DNS.

Also, from a code standpoint you never assign the service address if it parses correctly as an IP address, which means you're allowing the end user to pass in a DNS name but not an IP. Which probably wasn't the intent.